### PR TITLE
[ETP-297] Protect REST API Attendee better

### PR DESCRIPTION
**Pull request by Luc45**
Ticket: https://moderntribe.atlassian.net/browse/ETP-297

This PR prevents the usage of certain public endpoints that query Attendees when Event Tickets Plus is not loaded. The deleted tests were moved to the Event Tickets Plus repo.

I was purposefully vague on the changelog about our changes to which kind of access a user needs to have to read private Attendee data on REST, just in case. \m/

### CODE REVIEW

File: src/Tribe/REST/V1/Attendee_Repository.php [hunk 1]
```diff
@@ -39,7 +39,30 @@ public function __construct() {
...
+		 *
+		 * @param bool         $can_view_hidden_attendees The default value for whether the user can view hidden attendees.
+		 * @param null|WP_User $user                      The WP_User instance, if available.
+		 *
+		 * @since TBD
+		 *
+		 * @return bool Whether the current user can view hidden attendees.
+		 */
+		$can_view_hidden_attendees = (bool) apply_filters(
+			"tribe_tickets_user_can_view_hidden_attendees_rest",
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Attendee_Repository.php:60]: :no_entry_sign: String "tribe_tickets_user_can_view_hidden_attendees_rest" does not require double quotes; use single quotes instead

<sub>hash: 160447488450fb6f34fea1fb24602520</sub>

File: src/Tribe/REST/V1/Endpoints/Attendee_Archive.php [hunk 2]
```diff
@@ -81,9 +81,16 @@ public function get_documentation() {
 	 *
 	 * @param WP_REST_Request $request
 	 *
+	 * @since TBD Returns 401 Unauthorized if Event Tickets Plus is not loaded.
+	 *
 	 * @return WP_Error|WP_REST_Response An array containing the data on success or a WP_Error instance on failure.
 	 */
 	public function get( WP_REST_Request $request ) {
+		// Early bail: ET Plus must be active to use this endpoint.
+		if ( ! class_exists( "Tribe__Tickets_Plus__Main" ) ) {
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Endpoints/Attendee_Archive.php:90]: :no_entry_sign: String "Tribe__Tickets_Plus__Main" does not require double quotes; use single quotes instead

<sub>hash: 3eb3df06e5fd952bb1a55b069069f156</sub>

File: src/Tribe/REST/V1/Endpoints/Single_Attendee.php [hunk 3]
```diff
@@ -81,8 +81,15 @@ public function READ_args() {
 
 	/**
 	 * {@inheritdoc}
+	 *
+	 * @since TBD Returns 401 Unauthorized if Event Tickets Plus is not loaded.
 	 */
 	public function get( WP_REST_Request $request ) {
+		// Early bail: ET Plus must be active to use this endpoint.
+		if ( ! class_exists( "Tribe__Tickets_Plus__Main" ) ) {
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Endpoints/Single_Attendee.php:89]: :no_entry_sign: String "Tribe__Tickets_Plus__Main" does not require double quotes; use single quotes instead

<sub>hash: e35cbe35ffa3392d962fdd23f9c000c2</sub>

File: src/Tribe/REST/V1/Post_Repository.php [hunk 4]
```diff
@@ -88,8 +88,15 @@ public function get_data( $id, $context = '' ) {
 
 	/**
 	 * {@inheritdoc}
+	 *
+	 * @since TBD Returns 401 Unauthorized if Event Tickets Plus is not loaded.
 	 */
 	public function get_attendee_data( $attendee_id, $context = 'default' ) {
+		// Early bail: ET Plus must be active to get Attendee data.
+		if ( ! class_exists( "Tribe__Tickets_Plus__Main" ) ) {
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Post_Repository.php:96]: :no_entry_sign: String "Tribe__Tickets_Plus__Main" does not require double quotes; use single quotes instead

<sub>hash: 5e03be83d772af3968ca36859e4ac6b4</sub>

**Comment by bordoni** [comment]
Lets make sure tests are passing, but code looks good.

**Pull request merged**